File: PYTHON/Numeric-Functions/armstrong_between_ranges.py
def between_ranges(lower, upper):
	lower = int(lower)
	upper = int(upper)
	while (lower <= upper):
		results = is_armstrong(lower)
		if (results == int(1)):
			print(str(lower))
		lower = int(lower + 1)

def is_armstrong (num):
	num = int(num)
	orig = int(num)
	sums = int(0)
	d = digit_counter(num)
	while (num > 0):
		temp = int(num % 10)
		temp = pow(temp, d)
		sums = int(sums + temp)
		num = int(num / 10)
	if (int(sums) == int(orig)):
		return 1
	else:
		return 0

def digit_counter (num):
	num = int(num)
	counter = int(0)
	if (num == int(0)):
		counter = int(1)
	else:
		while (num > 0):
			counter = int(counter + 1)
			num = int(num / 10)
	return counter

lwr = input("User, enter the lower edge:\n")
upr = input("User, enter the upper edge:\n")

lwr = int(lwr)
upr = int(upr)

between_ranges(lwr, upr)

File: JAVA/armstrong_between_ranges.java
import java.util.Scanner;
import java.lang.Math;
class armstrong_between_ranges {
	public static void main (String [] args) {
		Scanner SC = new Scanner(System.in);
		int lwr;
		int upr;
		System.out.println("User, enter the lower edge: ");
		lwr = SC.nextInt();
		System.out.println("User, enter the upper edge: ");
		upr = SC.nextInt();
		between_ranges(lwr, upr);
	}
	public static void between_ranges (int lower, int upper) {
		while (lower <= upper) {
			int results = is_armstrong(lower);
			if (results == 1) {
				System.out.println(lower + " is an Armstrong");
			}
			lower++;
		}
		System.out.println("\nDONE!\n");
	}
	public static int is_armstrong (int num) {
		int orig = num;
		int d = digit_counter(num);
		int sums = 0;
		while (num > 0) {
			int temp = num % 10;
			temp = (int)Math.pow(temp,d);
			sums += temp;
			num /= 10;
		}
		if (sums == orig) {
			return 1;
		}
		else {
			return 0;
		}
	}
	public static int digit_counter (int num) {
		int counter = 0;
		if (num == 0) {
			counter = 1;
		}
		else {
			while (num > 0) {
				counter++;
				num /= 10;
			}
		}
	return counter;
	}
}

